reactive/archive_auth_mirror.py

```diff
@@ -1,14 +1,18 @@
 from charmhelpers.core import hookenv
 
-from charms.reactive import when, when_not
+from charms.reactive import when, when_not, hook
 
 from charms.layer.nginx import configure_site
 from charms.archive_auth_mirror import gpg, reprepro, utils
 
 
+@hook('install')
+def install():
+    utils.install_resources()
```

Comment:
Sure. But this isn't a normal hook.  If you would add a hook/install file, then you can be sure when that gets executed. But here you're telling reactive that you want this to be executed during the install hook.  Some of the layers probably also add handlers to be run during the install hook and trigger events.